x/pairing/keeper/unresponsive_provider.go

```diff
@@ -79,6 +80,19 @@ func (k Keeper) UnstakeUnresponsiveProviders(ctx sdk.Context, epochsNumToCheckCU
 	return nil
 }
 
+// getBlockEpochsAgo returns the block numEpochs back from the given blockHeight
+func (k Keeper) getBlockEpochsAgo(ctx sdk.Context, blockHeight uint64, numEpochs uint64) uint64 {
+	for counter := 0; counter < int(numEpochs); counter++ {
+		var err error
+		blockHeight, err = k.epochStorageKeeper.GetPreviousEpochStartForBlock(ctx, blockHeight)
+		if err != nil {
+			// too early in the chain life: bail without an error
+			return uint64(0)
```

Comment:
how about returning an error if it doesn't exist instead of 0? otherwise you are counting on the caller function to remember always to check for 0, which is more prone to errors than checking for err (that is enforced by go)